llvm/test/CodeGen/AMDGPU/live-interval-bug-in-rename-independent-subregs.mir

```diff
@@ -0,0 +1,475 @@
+  ; DeadInstDelete-NEXT:   renamable $sgpr0 = S_AND_B32 killed renamable $sgpr1, killed renamable $sgpr0, implicit-def dead $scc
+  ; DeadInstDelete-NEXT:   renamable $sgpr0 = S_AND_B32 killed renamable $sgpr0, 1, implicit-def dead $scc
+  ; DeadInstDelete-NEXT:   $scc = COPY killed renamable $sgpr0
+  ; DeadInstDelete-NEXT:   renamable $sgpr0 = S_CSELECT_B32 1, 0, implicit $scc
+  ; DeadInstDelete-NEXT:   renamable $sgpr0 = S_LSHL_B32 killed renamable $sgpr0, 1, implicit-def dead $scc
+  ; DeadInstDelete-NEXT:   renamable $sgpr0 = S_OR_B32 killed renamable $sgpr0, killed renamable $sgpr14, implicit-def dead $scc
+  ; DeadInstDelete-NEXT:   renamable $vgpr15 = COPY killed renamable $sgpr0
+  ; DeadInstDelete-NEXT:   S_BRANCH %bb.5
+  ; DeadInstDelete-NEXT: {{  $}}
+  ; DeadInstDelete-NEXT: bb.4:
+  ; DeadInstDelete-NEXT:   liveins: $sgpr4, $sgpr5, $sgpr6, $vgpr0, $vgpr15, $sgpr8_sgpr9:0x000000000000000F, $sgpr10_sgpr11
+  ; DeadInstDelete-NEXT: {{  $}}
+  ; DeadInstDelete-NEXT:   renamable $sgpr8_sgpr9 = S_LOAD_DWORDX2_IMM killed renamable $sgpr8_sgpr9, 0, 0 :: (invariant load (<2 x s32>), align 16, addrspace 4)
+  ; DeadInstDelete-NEXT:   renamable $vgpr1 = COPY killed renamable $sgpr5
+  ; DeadInstDelete-NEXT:   renamable $vgpr1_vgpr2, dead renamable $sgpr0_sgpr1 = V_MAD_U64_U32_e64 $sgpr9, killed $sgpr6, $vgpr1_vgpr2, 0, implicit $exec
+  ; DeadInstDelete-NEXT:   renamable $sgpr0_sgpr1_sgpr2_sgpr3 = S_LOAD_DWORDX4_IMM killed renamable $sgpr10_sgpr11, 0, 0 :: (invariant load (<4 x s32>), addrspace 4)
+  ; DeadInstDelete-NEXT:   renamable $sgpr5 = S_LSHL_B32 killed renamable $sgpr8, 6, implicit-def dead $scc
+  ; DeadInstDelete-NEXT:   renamable $vgpr1 = V_MUL_LO_U32_e64 killed $vgpr1, killed $sgpr5, implicit $exec
+  ; DeadInstDelete-NEXT:   renamable $sgpr4 = S_LSHL_B32 killed renamable $sgpr4, 6, implicit-def dead $scc
+  ; DeadInstDelete-NEXT:   renamable $vgpr0 = V_ADD_U32_e64 killed $sgpr4, killed $vgpr0, 0, implicit $exec
+  ; DeadInstDelete-NEXT:   renamable $vgpr0 = V_ADD_LSHL_U32_e64 killed $vgpr1, killed $vgpr0, 2, implicit $exec
+  ; DeadInstDelete-NEXT:   BUFFER_STORE_DWORD_OFFEN_exact killed renamable $vgpr15, killed renamable $vgpr0, killed renamable $sgpr0_sgpr1_sgpr2_sgpr3, 0, 0, 0, 0, implicit $exec :: (dereferenceable store (s32), align 1, addrspace 8)
+  ; DeadInstDelete-NEXT:   S_ENDPGM 0
+  ; DeadInstDelete-NEXT: {{  $}}
+  ; DeadInstDelete-NEXT: bb.5:
+  ; DeadInstDelete-NEXT:   successors: %bb.4(0x80000000)
+  ; DeadInstDelete-NEXT:   liveins: $sgpr4, $sgpr5, $sgpr6, $vgpr0, $vgpr15, $sgpr8_sgpr9:0x000000000000000F, $sgpr10_sgpr11, $sgpr12_sgpr13
+  ; DeadInstDelete-NEXT: {{  $}}
+  ; DeadInstDelete-NEXT:   $exec = S_OR_B64 $exec, killed renamable $sgpr12_sgpr13, implicit-def $scc
+  ; DeadInstDelete-NEXT:   S_BRANCH %bb.4
+  bb.39:
+    successors: %bb.82(0x40000000), %bb.40(0x40000000)
+    liveins: $sgpr0, $sgpr1, $sgpr2, $sgpr3, $sgpr4, $sgpr5, $sgpr6, $sgpr7, $vgpr0, $vgpr1, $vgpr2
+
+    %11:vgpr_32 = COPY $vgpr0
```

Comment:
Can you shrink this down any more? Also, use run-pass=none to compact the register numbers